Doc/library/unittest.mock-examples.rst

```diff
@@ -208,15 +208,13 @@ If you need an attribute setting on your mock, just do it:
     3
 
 Sometimes you want to mock up a more complex situation, like for example
-``mock.connection.cursor().execute("SELECT 1")``. If we wanted this call to
-return a list, then we have to configure the result of the nested call.
+``mock.connection.cursor().execute("SELECT 1")``.
 
 We can use :data:`call` to construct the set of calls in a "chained call" like
 this for easy assertion afterwards:
 
     >>> mock = Mock()
-    >>> cursor = mock.connection.cursor.return_value
-    >>> cursor.execute.return_value = ['foo']
+    >>> mock.connection.cursor.return_value.execute.return_value = ['foo']
```

Comment:
I'm not sure I think this is more clear, I do write my actual code this way but is it harder to read this way? Some new users struggle to reason about a line like this.

Reply:
I find this whole example somewhat confusing, since ISTM it's trying to explain two things at once:
1) how to mock a complex expression to return a certain value;
2) how to use `call` and `call_list()`,
I'm also not sure what the point of using `call` is, just to prove that both methods got called? If that's the case, is not very clear, and it does more harm than good IMHO.

The second line (`cursor = mock.connection.cursor.return_value`) is a bit unusual since we usually set the `return_value`, whereas here we are just getting a reference to the already existing default `return_value` (a mock).  This could be clarified.  I would just leave the shorthand suggested by Elena, and explain how it works.

To summarize, the whole thing could look like:
```rest
Sometimes you want to mock up a more complex expression, such as
``db.connection.cursor().execute("SELECT 1")``.  If we wanted this
call to return ``['foo']``, then we can do:

   >>> db = Mock()
   >>> db.connection.cursor.return_value.execute.return_value = ['foo']

Since we didn't specify a ``return_value`` for ``mock.connection.cursor()``,
a new mock will be created by default.  This new mock will have an
``execute`` method that returns ``['foo']`` when called::

   >>> cursor = db.connection.cursor()
   >>> cursor  # check that cursor is a new mock with an execute method
   <Mock name='db.connection.cursor()' id='140217681027816'>
   >>> cursor.execute("SELECT 1")
   ['foo']

This is equivalent to executing the initial expression::

   >>> db.connection.cursor().execute("SELECT 1")
   ['foo']
```

I would remove all the part about `call` -- if we want to explain how it works, we could do it in a separate entry.  I don't think in a real-world situation I would use it to check that both methods are called -- I would probably just check the final result and perhaps `db.connection.cursor.assert_called()`.  For this case, I don't think including these checks adds more value.